annotation_store/store.py

```diff
@@ -157,8 +157,10 @@ def _get_tool_provider(self):
 
         if 'launch_params' in self.request.LTI:
             params = self.request.LTI['launch_params']
-            return DjangoToolProvider(CONSUMER_KEY, lti_secret, params)
-        return DjangoToolProvider(CONSUMER_KEY, lti_secret)
+            return DjangoToolProvider.from_django_request(
```

Comment:
The following should go between line 159 and line 160 above. It will store the LTI params that are saved via middleware into the request's POST attribute in order to send the score back to the lti consumer.
```
            self.request.POST = self.request.POST.copy()
            for lti_params in params:
                self.request.POST[lti_params] = params[lti_params]
```